xbmc/utils/FileUtils.cpp

```diff
@@ -201,3 +203,59 @@ CDateTime CFileUtils::GetModificationDate(const std::string& strFileNameAndPath,
   }
   return dateAdded;
 }
+
+bool CFileUtils::CheckFileAccessAllowed(const std::string &filePath)
```

Comment:
Don't we already have  `CFileUtils::RemoteAccessAllowed` to do this?

Reply:
RemoteAccessAllowed doesn't properly validate the real path. I opted not to change that for v18, but add an additional check instead

Reply:
I don't see what is wrong with validation done by `RemoteAccessAllowed`, doesn't it just need calling earlier plus add the black list if we want one. If the additional check blocks more than `RemoteAccessAllowed` then it makes RemoteAccessAllowed redundant